lib/rucio/daemons/rsedecommissioner/profiles/generic.py

```
@@ -0,0 +1,483 @@
+    if (rule.expires_at is not None and rule.expires_at < datetime.utcnow()):
+        logger(logging.DEBUG,
+               'Rule %s for %s %s:%s is bound for deletion',
+               rule.id, rule.did_type, rule.scope, rule.name)
+        return 2
+
+    if rule['rse_expression'] == rse['rse']:
+        return 1
+
+    # Check the list of RSEs that the replicas locked by this rule reside on
+    locks = get_replica_locks_for_rule_id(rule.id, session=session)
+    replica_rse_ids = set(lock['rse_id'] for lock in locks)
+
+    if len(replica_rse_ids) == 1 and list(replica_rse_ids)[0] == rse['id']:
+        return 1
+
+    return 0
+
+
+@transactional_session
+def _generic_move_rule(
+    rule: models.ReplicationRule,
+    destination: str,
+    rse: 'Dict[str, Any]',
+    limit_check: LimitChecker,
+    *,
+    session: 'Session',
+    logger: 'Callable[..., None]' = logging.log,
+) -> None:
+    """Move the rules that satisfy deletion conditions to a specified destination.
+
+    :param rule: Rule to move.
+    :param destination: Move destination.
+    :param rse: RSE table entry as a dictionary.
+    :param limit_check: Throttling controller.
+    :param session: The database session in use.
+    :param logger: Logging function.
+    """
+    limit_check.processed_rules += 1
+
+    # Check the rule against various "filters".
+    if rule.child_rule_id:
+        logger(logging.DEBUG, 'Rule %s for %s %s:%s is being moved',
+               rule.id, rule.did_type.lower(), rule.scope, rule.name)
+
+        limit_check.increment(rule, logger=logger)
+        return
+
+    to_move = _check_removability(rule, rse, logger=logger, session=session)
+
+    if to_move == 1:
+        # Move the rule.
+        logger(logging.DEBUG, '(%s) Moving rule %s for %s %s:%s to %s',
+               rse['rse'], rule.id, rule.did_type.lower(),
+               rule.scope, rule.name, destination)
+
+        move_rule(rule.id, destination, session=session)
```

Add `override={'weight': None, 'source_replica_expression': None}` to avoid some potential issues.

Also, this could raise some exception and, as such, there needs to be some graceful handling of it somewhere. Similarly for `_generic_delete_rule()`.

---

What would be a recommended behavior? Should I catch all exceptions and let the daemon continue, or only RuleNotFound?